R/shared-methods.R

```diff
@@ -509,6 +509,53 @@ DataClass <- R6::R6Class(
         source_data_cols = paste(unlist(self$source_data_cols), collapse = ", ")
       )
       return(sum_df)
+    },
+
+    #' @description Specific test for a specific class.
+    #' By default this method is empty as if any code is required it should be
+    #' defined in a child class specific `specific_tests` method.
+    #' @param self_copy R6class the object to test
+    specific_tests = function(self_copy) {
```

Comment:
I am not sure that it makes sense to use this structure as opposed to the previous inheritance method. Can you clarify the advantage?

Reply:
Essentially I got into a knot with making a clone of the class in the parent `test` which would be called by the countries `test` (`super$test()`) which made it difficult to grab the cloned object used in the parent `test` for further tests. Doing it this way means I can just pass the cloned class to the specific tests function for their tests. Also helped with params as I don't have to redefine them all for specific tests, just params specific to that country if needed (e.g. `nhsregions` for the UK)